lib/smart_proxy_host_reports/ansible_processor.rb

```diff
@@ -82,46 +74,56 @@ def spool_report
+    elsif result["changed"]
+      add_keywords("HasChange")
     end
   end
 
-  def log_source(log)
-    # Temporary until  is resolved
-    log["module"]
+  def process_level(result)
+    if result["failed"]
+      result["level"] = "err"
+    elsif result["changed"]
+      result["level"] = "notice"
+    else
+      result["level"] = "info"
+    end
   end
 
-  def log_message(log)
-    return log["msg"] if log["failed"]
+  def process_message(result)
+    msg = "N/A"
+    return msg if result["task"].nil? || result["task"]["action"].nil?
+    return result["result"]["msg"] if result["failed"]
+    result_tree = result["result"]
+    task_tree = result["task"]
+    raise("Report do not contain required 'results/result' element") unless result_tree
+    raise("Report do not contain required 'results/task' element") unless task_tree
+    module_args_tree = result_tree.dig("invocation", "module_args")
 
-    case log["module"]
-    when "package"
-      log["results"].empty? ? log["msg"] : log["results"]
-    when "template"
-      module_args = log["invocation"]["module_args"]
-      "Rendered template #{module_args["_original_basename"]} to #{log["dest"]}"
-    when "service"
-      "Service #{log["name"]} #{log["state"]} (enabled: #{log["enabled"]})"
-    when "group"
-      "User group #{log["name"]} #{log["state"]}, gid: #{log["gid"]}"
-    when "user"
-      "User #{log["name"]} #{log["state"]}, uid: #{log["uid"]}"
-    when "cron"
-      module_args = log["invocation"]["module_args"]
-      "Cron job: #{module_args["minute"]} #{module_args["hour"]} #{module_args["day"]} #{module_args["month"]} #{module_args["weekday"]} #{module_args["job"]} (disabled: #{module_args["disabled"]})"
-    when "copy"
-      module_args = log["invocation"]["module_args"]
-      "Copy #{module_args["_original_basename"]} to #{log["dest"]}"
+    case task_tree["action"]
+    when "ansible.builtin.package"
+      detail = result_tree["results"] || result_tree["msg"] || "No details"
+      msg = "Package(s) #{module_args_tree["name"].join(",")} are #{module_args_tree["state"]}: #{detail}"
+    when "ansible.builtin.template"
+      msg = "Render template #{module_args_tree["_original_basename"]} to #{result_tree["dest"]}"
+    when "ansible.builtin.service"
+      msg = "Service #{result_tree["name"]} is #{result_tree["state"]} and enabled: #{result_tree["enabled"]}"
+    when "ansible.builtin.group"
+      msg = "User group #{result_tree["name"]} is #{result_tree["state"]} with gid: #{result_tree["gid"]}"
+    when "ansible.builtin.user"
+      msg = "User #{result_tree["name"]} is #{result_tree["state"]} with uid: #{result_tree["uid"]}"
+    when "ansible.builtin.cron"
+      msg = "Cron job: #{module_args_tree["minute"]} #{module_args_tree["hour"]} #{module_args_tree["day"]} #{module_args_tree["month"]} #{module_args_tree["weekday"]} #{module_args_tree["job"]} (disabled: #{module_args_tree["disabled"]})"
```

Comment:
```suggestion
      msg = "Cron job: #{module_args_tree["minute"]} #{module_args_tree["hour"]} #{module_args_tree["day"]} #{module_args_tree["month"]} #{module_args_tree["weekday"]} #{module_args_tree["job"]} and disabled: #{module_args_tree["disabled"]}"
```
If you're changing the way we had, let's stick with the same approach :)

Reply:
> If you're changing the way we had, let's stick with the same approach :)

Why? If we want to change anything, this is the right time. This is a complete rewrite, format is new, UI will be new. I am not following. I can do that, just explain the reason.

Reply:
Yes, that's OK. What I meant by that is that above you have no parentheses, but in this sentece you have them. My comment and suggestion is about to have consintency in the new way of friendly messages.

Reply:
Oh sure ok fixing.